include/seqan3/search/configuration/default_configuration.hpp

```diff
@@ -25,7 +25,10 @@ namespace seqan3::search_cfg
 /*!\brief The default configuration.
  * \ingroup search_configuration
  */
```

Comment:
Below suggestion will(should) fix the gcc7 errors.

`std::variant` should be `constexpr` in c++17, but in gcc7 it is not and it won't be fixed.

If this works, can you maybe add
```
\if DEV
\todo Make constexpr after GCC7 support is dropped.
\endif
```
to the documentation?